### DISCUSSION TITLE
Avoid using of mbedTLS

### DISCUSSION THREAD

**Discussion by vortigont** [mathieucarbou/ESPAsyncWebServer#34] [General]
Just to continue with our previous discussion

> platform = https://github.com/tasmota/platform-espressif32/releases/download/2024.05.00/platform-espressif32.zip
> It's a Tasmota special build of the Arduino library, based on Core 2.0.17 and IDF 4.4.7. It has some libs stopped out like mbedtsl, 

I've tried this one, it does indeed saves a lot of code size, but I did not get to use it properly. They removed crypto, but it brakes linking with any code that is related to `HTTPClient::begin`, does not matter it uses http or https.
So, @proddy any hints on how to use it for something other than Tasmota?
Actually what I need is to have HTTP Client/Server without any crypto junk.

**Comment by mathieucarbou** [comment]
I suspect something brings the mbedtls lib in Arduino 3, something related to the new refactoring of the Network / Wifi headers because the secured ones are not included by default and they depend on mbedtls.

When opening the ESPAsyncWebServer project and launching the PIO memory analyser we can see, when comparing the analysis of arduino-2 anv vs arduino-3 env that the lib is there only with Arduino 3.

**Comment by mathieucarbou** [comment]
I wish we could also add some define to completely disable the secured headers.

**Comment by vortigont** [comment]
I'm still using Arduino 2.x, mostly due to other libs incompatibilities and lack of official support for Core 3 on PIO.

> I wish we could also add some define to completely disable the secured headers.

Yeah! That strikes me too! Should not be that hard at all.
So I've started to strip-down official [HTTPClient](https://github.com/espressif/arduino-esp32/tree/master/libraries/HTTPClient) lib to create a light-weighted version that does not drag all TLS stuff along. It does help indeed, as I see from map file it still includes some crypto stuff that is required for WiFi, but removes all other SSL-related code and cert bundles. Results in 40Kb vs 120Kb.
Of course I had to remove all `begin()` overloads that uses SSL-related arguments, but at least it builds now and works as expected unless I feed some `https` URLs there.

Do not know why Tasmota can't do something like this, they already hacked into official Arduino core pretty hard.

**Comment by vortigont** [comment]
Made a PR for Arduino core espressif/arduino-esp32#9893
saves me about 80k of firmware image size and much more with Tasmota's core.
